src/model_2/data_loader.py

```
@@ -0,0 +1,71 @@
+        df = pd.read_csv(cache_path, index_variable='datetime')
+        df['datetime'] = pd.to_datetime(df['datetime'])
+        df.set_index('datetime', inplace=True)
+        return df
+
+    # 2. Fetch from UCI
+    logger.info("Fetching data from UCI repository...")
+    try:
+        beijing_multi_site_air_quality_data = fetch_ucirepo(id=501)
+        # Combine all features into one DataFrame
+        df = beijing_multi_site_air_quality_data.data.features
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)